lib/PAUSE/Indexer/Errors.pm

```diff
@@ -0,0 +1,121 @@
+package PAUSE::Indexer::Errors;
+use v5.12.0;
+use warnings;
+
+use Sub::Exporter -setup => {
+  exports => [ 'ERROR' ],
+  groups  => { default => [ 'ERROR' ] },
+};
+
+my %ERROR;
+
+sub public_error {
+  my ($name, $arg) = @_;
+  $ERROR{$name} = {
+    ident   => $name,
+    public  => 1,
+    %$arg,
+  };
+}
+
+public_error blib => {
+  header  => 'archive contains a "blib" directory',
+  body    => <<~'EOF'
+    The distribution contains a blib/ directory and is therefore not being
+    indexed.  Hint: try 'make dist'.
+    EOF
+};
+
+public_error multiroot => {
+  header  => 'archive has multiple roots',
+  body    => sub {
+    my ($dist) = @_;
+    return <<~"EOF"
+      The distribution does not unpack into a single directory and is therefore
+      not being indexed. Hint: try 'make dist' or 'Build dist'. (The directory
+      entries found were: @{$dist->{HAS_MULTIPLE_ROOT}})
+    EOF
+  },
+};
+
+public_error no_distname_permission => {
+  header  => 'missing permissions on distname package',
+  body    => sub {
+    my ($dist) = @_;
+
+    my $pkg = $dist->_package_governing_permission;
+
+    return <<~"EOF"
```

Comment:
NO_DISTNAME_PERMISSION used to talk about:
```
   push @m, $tf->format(qq[This distribution name will only be indexed
      when uploaded by users with permission for the package $pkg.
      Either someone else has ownership over that package name, or
      this is a brand new distribution and that package name was neither
      listed in the 'provides' field in the META file nor found
      inside the distribution's modules.  Therefore, no modules
      will be indexed.]);
```

And when...
 `        # Perhaps they forgot a pm file matching the dist name`
...then it would say the paragraph you have below.

Why is this changing here to no longer have the former case? What happens in the new code if I upload a dist someone else already owns but I do have a matching .pm name for the dist?

Reply:
Restored!